toast-tk-runtime/src/main/java/io/toast/tk/runtime/report/MailReportSender.java

```diff
@@ -0,0 +1,146 @@
+
+		final Properties properties = new Properties();
+		try (final Reader resourceFileReader = new InputStreamReader(resourceAsStream)) {
+			properties.load(resourceFileReader);
+		} catch (final IOException e) {
+			LOG.error("Could not load smtp.properties file", e);
+		}
+
+		if (Boolean.parseBoolean(properties.getProperty
+				("mail.smtp.auth"))) {
+			return Session.getDefaultInstance(properties, new Authenticator() {
+				@Override
+				protected PasswordAuthentication getPasswordAuthentication() {
+					String user = properties.getProperty("mail.smtp.user");
+					String password = properties.getProperty("mail.smtp.password");
+					return new PasswordAuthentication(user, password);
+				}
+			});
+		} else {
+			return Session.getDefaultInstance(properties);
+		}
+	}
+
+	private AdaptersConfig getConfig() {
+		return new AdaptersConfigProvider().get();
+	}
+
+	private void send(ICampaign campaign, List<String> mailTo, String mailFrom, Session session) {
+		try {
+			MimeMessage message = new MimeMessage(session);
+			message.setFrom(new InternetAddress(mailFrom));
+			InternetAddress address = new InternetAddress();
+			for (String recipient : mailTo) {
+				address.setAddress(recipient);
+				message.addRecipient(Message.RecipientType.TO, address);
+			}
+			message.setSubject(getSubject());
+			message.setText(getMailBody(campaign));
+
+			Transport.send(message);
+		} catch (MessagingException e) {
+			LOG.error(e.getMessage(), e);
+		}
+	}
+
+	private String getSubject() {
+		return "Test report " + getDateAsString(LocalDateTime.now());
+	}
+
+	private String getDateAsString(LocalDateTime localDateTime) {
+		return localDateTime.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM));
+	}
+
+	private String getMailBody(ICampaign campaign) {
+		StrBuilder sb = new StrBuilder();
+
+		sb.append("Test report for the test campaign: ").append(campaign.getName());
+		sb.appendNewLine();
+
+		// First test start time
+		if (CollectionUtils.isNotEmpty(campaign.getTestCases())) {
+			long startDateMillis = campaign.getTestCases().get(0).getStartDateTime();
+			LocalDateTime startDateTime =
+					LocalDateTime.ofInstant(Instant.ofEpochMilli(startDateMillis), ZoneId.systemDefault());
+			sb.append("Start time: ").append(getDateAsString(startDateTime)).appendNewLine();
+		}
+		sb.appendNewLine();
+
+		// Success number
+		sb.append(getSuccessNumber(campaign)).append(" tests in success").appendNewLine();
+
+		// Failure number
+		long failureNumber = getFailureNumber(campaign);
+		sb.append(failureNumber).append(" tests failed").appendNewLine();
+		sb.appendNewLine();
+
+		if (failureNumber > 0) {
+			sb.append("Failed test cases:").appendNewLine();
+
+			for (final ITestPage testPage : campaign.getTestCases()) {
+				if (!testPage.isSuccess()) {
+					sb.append("- ").append(testPage.getName()).appendNewLine();
+				}
+			}
+		}
+		String s = sb.toString();
+		System.out.println("s = " + s);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)